ckanext/harvest/logic/action/update.py

```diff
@@ -225,6 +225,68 @@ def harvest_source_clear(context, data_dict):
     return {'id': harvest_source_id}
 
 
+def harvest_sources_job_history_clear(context, data_dict):
+    '''
+    Clears the history for all active harvest sources. All jobs and objects related to a harvest source will
+    be cleared, but keeps the source itself.
+    This is useful to clean history of long running harvest sources to start again fresh.
+    The datasets imported from the harvest source will NOT be deleted!!!
+
+    '''
+    check_access('harvest_sources_clear', context, data_dict)
+
+    job_history_clear_results = []
+    # We assume that the maximum of 1000 (hard limit) rows should be enough
+    result = logic.get_action('package_search')(context, {'fq': '+type:"harvest"', 'rows': 1000})
```

Comment:
The Solr index field that holds the dataset type is called `dataset_type`, so this query does no return anything. You should change it to `{'fq': '+dataset_type:harvest', 'rows': 1000})`. Note that if you want to wrap the value in double quotes we need to this change in plugin.py:

```diff
diff --git a/ckanext/harvest/plugin.py b/ckanext/harvest/plugin.py
index 55af1c7..fa57103 100644
--- a/ckanext/harvest/plugin.py
+++ b/ckanext/harvest/plugin.py
@@ -92,7 +92,7 @@ class Harvest(p.SingletonPlugin, DefaultDatasetForm, DefaultTranslation):
         '''Prevents the harvesters being shown in dataset search results.'''
 
         fq = search_params.get('fq', '')
-        if 'dataset_type:harvest' not in fq:
+        if 'dataset_type:harvest' not in fq and 'dataset_type:"harvest"' not in fq:
             fq = u"{0} -dataset_type:harvest".format(search_params.get('fq', ''))
             search_params.update({'fq': fq})
 
```
Which is an excellent idea anyway.